### PR TITLE
Fade in lists one by one

### PR DESCRIPTION
**Pull request by vonElfvin**
Kapture 2020-10-29 at 20 54 13
Kapture 2020-10-29 at 20 59 53

### CODE REVIEW

**Review by rasmusguterstam — APPROVED** [review]
Let's goo 🔥

File: shared/hedvig-ui/animations/fade-in.tsx [hunk 1]
```diff
@@ -6,10 +6,11 @@ const fadeIn = (max) =>
     to: { opacity: max, transform: 'translateY(0)' },
   })
 
-export const FadeIn = styled('div')<{ delay?: string }>(
-  ({ delay = '0ms' }) => ({
+export const FadeInComponent = (component: any) =>
```
Comment (palmenhq) [inline shared/hedvig-ui/animations/fade-in.tsx:9]: Super nice and reusable! But a) i'd call this maybe something like `withFadeIn` or maybe actually just `fadeIn` (and rename the keyframes to `fadeInKeyframes` or so), the capital F indicates it'd be a component itsself which is not true - it's just a decorator :) and b) the typing needs some work.

Re the typing: Maybe we should try to "decorate" the styled function a bit more extensively. I think that'd be easiest considering emotions' typings being quite advanced. But i guess you could do something like:

```ts
const fadeIn: CreateStyledFunction<object> = (component, ...args) => styled(component, ...args)<{ delay?: string }>(({ delay = '0ms' }) => ({ 
...
```

**Review by palmenhq — COMMENTED** [review]
awesome!

**Review by palmenhq — APPROVED** [review]
awesome! 👑

**Pull request merged**